Problem_2.java

```diff
@@ -0,0 +1,27 @@
+// Time Complexity : O(n), n= no. of nodes in tree
```

Comment:
Since you are running a `for` loop at each recursive call, this algorithm won't run in linear time.